applications/matter_weather_station/Kconfig

```diff
@@ -13,4 +13,10 @@ config AVERAGE_CURRENT_CONSUMPTION
 	  The predicted average current consumption of the Matter weather station
 	  device, used to estimate the remaining battery time.
 
+# Overwrite the Fprotect block size to fit the size of the factory data partition.
```

Comment:
I think this is actually wrong. I think this Kconfig was meant to be descriptive only and expose the hardware limitation and I doubt changing it changes the way the hardware works. You can double-check, but I think we can't do it this way.

Reply:
@ArekBalysNordic claims it was added because of the factory data partition size, so I added check to do that only for factory data build type. Arek, could you elaborate more on that?